install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml

```diff
@@ -142,10 +142,10 @@ data:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
-        - name: ISTIO_META_POD_NAME
+        - name: ISTIO_META_CONFIG_NAMESPACE
           valueFrom:
             fieldRef:
-              fieldPath: metadata.name
+              fieldPath: metadata.namespace
```

Comment:
Also: shorter keys may be better since this is sent on each request, right now metadata is probably > 1/2 of the pilot traffic and allocs ( with incremental EDS enabled most pushes without config changes are pretty small - but the ack/nack including all meta is larger than the actual payload ).

Reply:
i.e. ISTIO_META_NS

Reply:
This is just config namespace. The pod name is irrelevant here. I could choose to put arbitrary set of pods, VMs in one single config namespace (does not have to align with K8S namespace, though it does today). And the code extracts it as well.

As for ISTIO_META_NS, I thought about it but decided against it as all other variables are fully named (like INTERCEPTION_MODE, METAJSON_ANNOTATIONS, etc.). Also, i realized that when people grok at the kube-injected yaml file, having something readable really helps.